h5grove/content.py

```diff
@@ -233,6 +233,19 @@ def metadata(self, depth: int = 1):
         )
 
 
+class DatatypeContent(ResolvedEntityContent[h5py.Datatype]):
+    kind = "datatype"
+
+    def metadata(self, depth=None):
+        """
+        :returns: {"attributes": AttributeMetadata, "kind": str, "name": str, "type": TypeMetadata}
```

Comment:
Since our minimal Python version is 3.8, we can now make this proper type hints with [TypedDict](https://docs.python.org/3/library/typing.html#typing.TypedDict)

Reply:
Good idea, I'll try to do it in a separate PR.